lib/saml_idp/response_builder.rb

```diff
@@ -1,18 +1,35 @@
 require 'builder'
 module SamlIdp
   class ResponseBuilder
+    include Signable
+
     attr_accessor :response_id
     attr_accessor :issuer_uri
     attr_accessor :saml_acs_url
     attr_accessor :saml_request_id
     attr_accessor :assertion_and_signature
+    attr_accessor :algorithm
 
-    def initialize(response_id, issuer_uri, saml_acs_url, saml_request_id, assertion_and_signature)
+    def initialize(
+      response_id,
+      issuer_uri,
+      saml_acs_url,
+      saml_request_id,
+      assertion_and_signature,
+      algorithm,
+      x509_certificate,
+      secret_key,
+      cloudhsm_key_label
```

Comment:
this is getting to be a lot of ordered params, do you think it would be worth switching to keyword arguments so it's harder to get things wrong?

Reply:
I agree... but I also feel that we should avoid refactoring of this gem if at all possible since we forked it and probably should at some point try to bring in updates made upstream. The fewer changes the better. Which may be an argument to undo my rubocop fixes ;-)

Reply:
Oh right I forgot we forked this. Fair enough!

Reply:
Oh right I forgot we forked this. Fair enough!